js/payment-methods/bitcoin.js

```diff
@@ -217,35 +224,150 @@ app.paymentMethods.bitcoin = (function() {
+						}, function(error, results) {
+							if (error) {
+								tryEachCallback(error);
+							} else {
+								tryEachCallback(null, results);
+							}
+						})
+					}
+				],
+				function(error, results) {
+
+					if (error) {
+						return cb(error);
+					}
+
+					var wasReceived = results[0];
+					var amountReceived = results[1];
+
+					cb(null, wasReceived, amountReceived);
+				});
+
+
+			}, this));
+		},
+
+		checkIfIndexWasUsed: function(xpub, index, cb) {
+
+			this.getAddress(index, xpub, _.bind(function(error, address) {
+
+				if (error) {
+					cb(error);
 				}
 
-				uri += '/api/addr/' + encodeURIComponent(address) + '/unconfirmedBalance';
+				var networkName = this.getNetwork(xpub).name;
+
+				async.tryEach([
+					function getTotalReceivedFromBlockexplorer(tryEachCallback) {
+						async.retry(app.config.retryFunctionsConfig, function(retryCallback) {
+							app.services.blockexplorer.getTotalReceiveByAddressAndNetworkName(address, networkName, retryCallback);
+						}, function(error, results) {
+
+							if (error) {
+								tryEachCallback(error);
+							} else {
+								tryEachCallback(null, results);
+							}
+						})
+					}
+				],
+				function(error, results) {
+
+					if (error) {
+						return cb(error);
+					}
 
-				$.get(uri).then(function(result) {
+					var totalReceived = results;
 
 					try {
-						var amountReceived = new BigNumber(result);
-						// Convert to BTC from satoshis.
-						amountReceived = amountReceived.dividedBy('100000000');
+						var indexWasUsed = totalReceived.greaterThan('0');
 					} catch (error) {
 						return cb(error);
 					}
 
-					var wasReceived = amountReceived.greaterThanOrEqualTo(amount);
-					cb(null, wasReceived, amountReceived);
+					cb(null, indexWasUsed);
+				})
 
-				}).fail(cb);
+			}, this))
+		},
+
+		/**
+			finds the first index that contains gapLimit number of not used addresses
+		 */
+
+		getGapLimit: function(xpub, cb) {
+			var gapLimitRequired = this.gapLImit;
+			var gapLimit = 0;
+			var index = 0;
+			var firstIndexOfGap = index;
+			var checkIfIndexWasUsed = _.bind(this.checkIfIndexWasUsed, this);
+			var ref = this.ref;
+			
+			async.whilst(
+				function() { return gapLimit < gapLimitRequired },
+				function(callback) {
+					checkIfIndexWasUsed(xpub, index, function(error, indexWasUsed) {
+
+						if (error) {
+							console.log('error', error);
+							callback(error);
+						} else if (indexWasUsed) {
+							gapLimit = 0;
+							index++;
+							firstIndexOfGap = index;
+							callback(null, 1);
+						} else {
+							gapLimit++;
+							index++;
+							callback(null, gapLimit);
+						}
+
+					})
+
+				},
+				function(error, n) {
+					var settingsUpdatObj = {};
+
+					if (error) {
+						console.log(app.i18n.t(ref + 'settings.error-getting-gap-limit'));
+						return cb(error);
+					}
+
+					cb(null, firstIndexOfGap);
+
+				}
+			)
+
+		},
+
+		validateAsync: function(data, cb) {
```

Comment:
Validate function should be per setting - not per payment method (see [here](https://github.com/Learn-by-doing/crypto-terminal/blob/d7a88a4e2d6a2a1b2d103d322cccf2c243189f4f/js/payment-methods/bitcoin.js#L50-L54)).